### PR TITLE
unpack bits function

### PR DESCRIPTION
**Pull request by havardpe**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@arnej27959 please review
@lesters FYI

### CODE REVIEW

File: eval/src/vespa/eval/instruction/unpack_bits_function.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * Tensor function unpacking bits into separate values.
+ *
+ * Both the tensor containing the packed bits and the result tensor
+ * must have cell type 'int8'. The bits must be unpacked in canonical
+ * order; bytes are unpacked with increasing index, bits within a byte
+ * are unpacked from most to least significant.
+ *
+ * The baseline expression looks like this:
+ *
+ * tensor<int8>(x[64])(bit(packed{x:(x/8)},7-(x%8)))
```
Comment (arnej27959) [inline eval/src/vespa/eval/instruction/unpack_bits_function.h:19]: My guess is that users will write
`tensor<int8>(x[64])(bit(packed{x:(x/8)},x%8))`
instead...

**Review by arnej27959 — APPROVED** [review]
looks good, I just hope we can write some really good documentation on how to use this.

File: eval/src/vespa/eval/instruction/unpack_bits_function.h (hunk 1 above)
Comment (arnej27959) [reply eval/src/vespa/eval/instruction/unpack_bits_function.h:19]: On second thought they will probably write
`tensor(x[64])(bit(packed{x:(x/8)},x%8))`

Comment (havardpe) [reply eval/src/vespa/eval/instruction/unpack_bits_function.h:19]: I have considered adding support for 'little endian' as well (top alternative form), but wanted to do the basic expression first. I have checked the unpack_bits function from python, and that one defaults to 'big endian'. The last alternative form could also be recognized if we go for 'any cell type as output', which would actually by quite reasonable. A possible solution is to add an unpack_bits second order function in Java that will expand to the appropriate expression, I am not sure how complicated that would be. @lesters any thoughts?

**Pull request merged**